### PR TITLE
feat: add useFileInput hook

### PR DESCRIPTION
**Pull request by korvin89**

### CODE REVIEW

File: src/components/utils/useFileInput/__stories__/Select.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
```
Comment (amje) [inline src/components/utils/useFileInput/__stories__/Select.stories.tsx:1]: Filename is Select.stories.tsx

File: src/components/utils/useFileInput/useFileInput.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            onChange?.(event);
+            onUpdate?.(Array.from(event.target.files || []));
+            // https://stackoverflow.com/a/12102992
+            event.target.value = '';
+        },
+        [onChange, onUpdate],
+    );
+
+    return {
+        controlProps: {
```
Comment (amje) [inline src/components/utils/useFileInput/useFileInput.ts:57]: Let's memo this object?

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/utils/useFileInput/useFileInput.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        return (
+            <React.Fragment>
+                <input ref={ref} {...controlProps} />
+                <Button onClick={openDeviceStorage}>Upload</Button>
+            </React.Fragment>
+        );
+    };
+```
+*/
+export function useFileInput({ref, onUpdate, onChange}: UseFileInputProps): UseFileInputOutput {
```
Comment (ValeraS) [inline src/components/utils/useFileInput/useFileInput.ts:41]: Maybe it's better to return the ref from the hook?

File: src/components/utils/useFileInput/useFileInput.ts (hunk 2 above)
Comment (ValeraS) [reply src/components/utils/useFileInput/useFileInput.ts:57]: Why do it?

File: src/components/utils/useFileInput/useFileInput.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    );
+
+    return {
+        controlProps: {
+            type: 'file',
+            tabIndex: -1,
+            style: {opacity: 0, position: 'absolute', width: 1, height: 1},
+            onChange: handleChange,
+        },
+        openDeviceStorage,
```
Comment (ValeraS) [inline src/components/utils/useFileInput/useFileInput.ts:63]: let's make like with controlProps:
```suggestion
        triggerProps: {
            onClick: openDeviceStorage,
        },
```

**Review by amje — APPROVED** [review]

**Pull request merged**